recipes/namedtype/all/conanfile.py

```diff
@@ -17,10 +20,40 @@ def _source_subfolder(self):
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
-        url = self.conan_data["sources"][self.version]["url"]
-        extracted_dir = "NamedType-" + os.path.splitext(os.path.basename(url))[0]
+        extracted_dir = glob.glob("NamedType-*")[0]
         os.rename(extracted_dir, self._source_subfolder)
 
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+
     def package(self):
-        self.copy("*.hpp", dst="include", src=self._source_subfolder)
+        if self.version == "20190324":
+            self.copy("*.hpp", dst="include/NamedType", src=self._source_subfolder)
+        else:
+            self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, 'include'))
```

Comment:
```suggestion
        if self.version == "20190324":
            self.copy("*.hpp", dst=ps.path.join("include", "NamedType"), src=self._source_subfolder)
        else:
            self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, 'include'))
```

In `package_info()`, I would append `os.path.join("include", "NamedType")` in `cpp_info.includedirs` if `self.version == "20190324"`, to avoid breaking downstreams.

Reply:
Agree